gradoop-flink/src/main/java/org/gradoop/flink/model/api/epgm/BaseGraphOperators.java

```diff
@@ -0,0 +1,247 @@
+   * result of those functions at the resulting graph using the given property keys.
+   *
+   * @param aggregateFunctions computes aggregates on the logical graph
+   * @return logical graph with additional properties storing the aggregates
+   */
+  LG aggregate(AggregateFunction... aggregateFunctions);
+
+  /**
+   * Sets the aggregation result of the given function as property for each vertex. All edges where
+   * the vertex is relevant get joined first and then grouped. The relevant edges are specified
+   * using the direction which may direct to the vertex, or from the vertex or both.
+   *
+   * @param function      aggregate function
+   * @param edgeDirection incoming, outgoing edges or both
+   * @return logical graph where vertices store aggregated information about connected edges
+   */
+  LG reduceOnEdges(EdgeAggregateFunction function, Neighborhood.EdgeDirection edgeDirection);
+
+  /**
+   * Sets the aggregation result of the given function as property for each vertex. All vertices
+   * of relevant edges get joined first and then grouped by the vertex. The relevant edges are
+   * specified using the direction which may direct to the vertex, or from the vertex or both.
+   *
+   * @param function      aggregate function
+   * @param edgeDirection incoming, outgoing edges or both
+   * @return logical graph where vertices store aggregated information about connected vertices
+   */
+  LG reduceOnNeighbors(VertexAggregateFunction function, Neighborhood.EdgeDirection edgeDirection);
+
+  /**
+   * Verifies this graph, removing dangling edges, i.e. edges pointing to or from
+   * a vertex not contained in this graph.<br>
+   * This operator can be applied after an operator that has not checked the graphs validity.
+   * The graph head of this logical graph remains unchanged.
+   *
+   * @return this graph with all dangling edges removed.
+   */
+  LG verify();
+
+  //----------------------------------------------------------------------------
+  // Binary Operators
+  //----------------------------------------------------------------------------
+
+  /**
+   * Creates a new logical graph by combining the vertex and edge sets of
+   * this graph and the given graph. Vertex and edge equality is based on their identifiers.
+   *
+   * @param otherGraph logical graph to combine this graph with
+   * @return logical graph containing all vertices and edges of the input graphs
+   */
+  LG combine(LG otherGraph);
+
+  /**
+   * Creates a new logical graph containing the overlapping vertex and edge sets of this graph
+   * and the given graph. Vertex and edge equality is based on their identifiers.
+   *
+   * @param otherGraph logical graph to compute overlap with
+   * @return logical graph that contains all vertices and edges that exist in
+   * both input graphs
+   */
+  LG overlap(LG otherGraph);
+
+  /**
+   * Creates a new logical graph containing only vertices and edges that
```

Comment:
Maybe add a comment on what happens with the graph head here, since graph heads are kept with this operator.